frontend/src/metabase-lib/v1/expressions/resolver.js

```diff
@@ -40,32 +40,29 @@ function findMBQL(op) {
   return clause;
 }
 
-// a is the type of the argument expected,
-// as defined in MBQL_CLAUSES,
-// and b is the inferred type of the argument
-const isCompatible = (a, b) => {
-  if (a === "any") {
+const isCompatible = (expectedType, inferredType) => {
+  if (expectedType === "any" || inferredType === "any") {
```

Comment:
The second part of the condition is new, the rest of the function works the same as before